application-analytics-ui/src/main/resources/Analytics/Code/Macros/VisitsSummary.xml

```diff
@@ -220,45 +228,46 @@ require(['jquery', 'chart'], function ($) {
   }
 
   // Updates the chart.
-  function editCanvas(canvas, cacheId, date, endpoints, fields, period, yAxeNames) {
+  function editCanvas(canvas, cacheId, date, endpoints, fields, period, axis) {
     let requestPageRef = XWiki.Model.resolve('Analytics.Code.Macros.VisitsSummaryJson', XWiki.EntityType.DOCUMENT);
     let url = new XWiki.Document(requestPageRef).getURL('get');
-    let requests = []; // This is correctly defined
-    // A field can have multiple endpoints(lines) call all of them and store the promise in vector for later use.
-    for (let i = 0; i &lt; endpoints.length; i++) {
+    let requestsPromises = [];
+    // A field can have multiple endpoints(lines in the graph eg Unique Views and Views) call all of them and store the promise in vector for later use.
+    endpoints.forEach(endopint =&gt; {
       let queryString = $.param({
         'date': date,
-        'apiMethod': endpoints[i],
+        'apiMethod': endopint,
         'period': period
       });
-      requests.push(makeRequest(url, queryString));
-    }
+      requestsPromises.push(makeRequest(url, queryString));
+    });
     // After all the promises are completed update the chart with the new data.
-    Promise.all(requests).then(responses =&gt; {
+    Promise.all(requestsPromises).then(responses =&gt; {
       let chart = chartsMap.get(cacheId);
       if (chart != undefined) {
         chart.destroy();
       }
       let values = fields.map((field, i) =&gt; responses[i][field]);
       // Map the chart to the id of the macro to be able to edit the canvas.
-      chartsMap.set(cacheId, createChart(canvas, responses[0]["labels"], values, yAxeNames));
+      chartsMap.set(cacheId, createChart(canvas, responses[0]['labels'], values, axis));
       canvas.removeClass("loading");
     }).catch(error =&gt; {
-      console.error("Request failed while in the VisitsSummary macros:", error)
+      new XWiki.widgets.Notification(l10n['error'], 'error');
+      console.error('Request failed while in the VisitsSummary macros:', error)
```

Comment:
You should leave a space after the written message, to have it nice displayed
`... VisitsSummary macros: ', ...`